lib/erb/compiler.rb

```diff
@@ -381,6 +381,8 @@ def compile_content(stag, out)
       else
         out.push(content)
       end
+    when /([ \t]*)<%-=/
+      add_insert_cmd(out, "-> (c) { c.to_s.empty? ? '' : '#{$1}' + c }.((#{content}))")
```

Comment:
Calling a Proc and branching with `c.to_s.empty?` on every `<-=` is very suboptimal. I'm reluctant to merge this feature as long as it requires this since it will be significantly harder to maintain the current performance with this. Can you revise the design to eliminate this dynamic code on rendering?

If the feature definitely needs this, I'd like to make this a third-party gem. We've supported adding an extra syntax before, so we can do that now.

Reply:
Since whether to add the prefix of spaces depends on the result of content, the only other way I can think of is to define a `String.add_prefix_if_present` method. But it's more ugly..

```ruby
class String
  def add_prefix_if_present(prefix)
    self == '' ? self : prefix + self
  end
end

add_insert_cmd(out, "(#{content}).to_s.add_prefix_if_present($1)")
```

Any other idea or would you like to implement it by your will?

Reply:
> Any other idea or would you like to implement it by your will?

Sorry, I don't have bandwidth to do that for you. In the meantime, let's maintain this feature at https://github.com/k0kubun/erb-trim. Once we figure out a way to resolve it at compile time, we can revisit this at ruby/erb.

Reply:
I plan to revisit this a month later or so.